### PR TITLE
LPS-146529 translation error download CSV

### PR DESCRIPTION
**Pull request by boton**
Issue: https://issues.liferay.com/browse/LPS-146529

### CODE REVIEW

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java [hunk 1]
```diff
@@ -38,6 +38,10 @@ public int doStartTag() throws JspException {
 
 		setContainerElement("a");
 
+		if (Validator.isNotNull(_download)) {
+			setDynamicAttribute(StringPool.BLANK, "download", _download);
+		}
+
```
Comment (adolfopa) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: We'll need to send this PR to FI for further review (unless you've already agreed on this with them).

File: modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/ImportTranslationResultsDisplayContext.java [hunk 2]
```diff
@@ -66,6 +73,35 @@ public int getFailureMessagesCount() {
...
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception, exception);
+			}
+		}
+
+		return "data:text/csv;charset=utf-8," + JS.encodeURIComponent(csv);
+	}
+
```
Comment (adolfopa) [inline modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/ImportTranslationResultsDisplayContext.java:104]: If it fails, let it fail naturally, don't swallow errors we cannot handle;  returning an empty value on failure looks wrong to me. Expose the exception type (not Exception; as a last resort use PortalException, wrapping the original failure).

Also, avoid having unnecessary variables. If you let it fail, you can directly return the result of `stringWriter.toString()`.

File: modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationResultsMVCRenderCommand.java [hunk 3]
```diff
@@ -56,12 +63,26 @@ public String render(
 			httpSession.getAttribute(
 				ImportTranslationResultsDisplayContext.class.getName()));
 
+		renderRequest.setAttribute(
+			FFBulkTranslationConfiguration.class.getName(),
+			_ffBulkTranslationConfiguration);
+
```
Comment (adolfopa) [inline modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationResultsMVCRenderCommand.java:69]: In general, try to inject as few request variables as possible. It is better to inject the configuration into the display context (so you need to add the dependency where it is created, in `ImportTranslationMVCActionCommand`), and expose a new method in the display context named `isDownloadCSVReportEnabled()` or something like that.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java (hunk 1 above)
Comment (boton) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: Oh sure, I can mention them to check this commit :D

File: modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationResultsMVCRenderCommand.java (hunk 3 above)
Comment (boton) [reply modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/ImportTranslationResultsMVCRenderCommand.java:69]: I'll do that

File: modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/ImportTranslationResultsDisplayContext.java (hunk 2 above)
Comment (boton) [reply modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/ImportTranslationResultsDisplayContext.java:104]: That makes sense, @AliciaGarciaGarcia I'll need your help with this :D

**Comment by boton** [comment]
Hi @liferay-lima, I need to use the download attribute in the <clay:link /> taglib, it is added in the [TLD](https://github.com/liferay/liferay-portal/blob/d2f48da96be1919b5656864a4619c22ed554eeee/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld#L1543-L1548) but isn't implemented.

Can you review if I implemented it 8c192f764ab1967d64a17e814c4709b8178caddd in a good way?

Thanks in advance!

**Comment by boton** [comment]
Hi @liferay-frontend, I need to use the download attribute in the <clay:link /> taglib, it is added in the [TLD](https://github.com/liferay/liferay-portal/blob/d2f48da96be1919b5656864a4619c22ed554eeee/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld#L1543-L1548) but isn't implemented.

Can you review if I implemented it 8c192f764ab1967d64a17e814c4709b8178caddd in a good way?

Thanks in advance!

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
Everything looks good to me. Not forwarding; will wait for FI response (feel free to forward it yourself when done).

**Comment by julien** [comment]
Just pinging @carloslancha and @tinycarol to make sure they see this.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java (hunk 1 above)
Comment (carloslancha) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: LGTM, just check that is being rendered correctly in React when the Tag is hydrated (if you pass some additionalProps or propsTransformer it's automatically hydrated)

Comment (boton) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: Sorry, @carloslancha I've added the two attributes but nothing is rendered by React only the plain additional attributes.

image
image

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java (hunk 1 above)
Comment (boton) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: Hey, @carloslancha, did you have time to review this? It is enough to send as is now.

**Comment by boton** [comment]
Rebased to avoid conflicts in bnd

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java (hunk 1 above)
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: @adolfopa apparently @carloslancha was mostly OK with that change

> LGTM, just check that is being rendered correctly in React when the Tag is hydrated (if you pass some additionalProps or propsTransformer it's automatically hydrated)

Did you manage to check this @boton? Do you need help?

Comment (boton) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java:44]: @julien Yep I tested here but didn't work 

> Sorry, @carloslancha I've added the two attributes but nothing is rendered by React only the plain additional attributes.
> 
> image image

I think the hydration in react isn't merged yet liferay-frontend#1801

**Comment by boton** [comment]
Rebased avoiding conflicts

**Comment by boton** [comment]
ci:forward

**Comment by boton** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:forward

**Comment by carloslancha** [comment]
Hey guys! I'm not sure what's going on with my gh config but I didn't receive any notification about this, sorry!! If you notice a delay like this one once again please feel free to reach me via slack 🙏 

Thanks y'all!!

**Comment by boton** [comment]
Don't worry, it is on review at FI https://github.com/liferay-frontend/liferay-portal/pull/1939. We will reach you via slack if it happens in the future.

**Pull request closed without merging**